rplugin/python3/deoplete/sources/minisnip.py

```diff
@@ -9,14 +11,25 @@ def __init__(self, vim):
         self.name = 'minisnip'
         self.mark = '[minisnip]'
         self.min_pattern_length = 0
-        self.minisnip_dir = self.vim.eval('g:minisnip_dir')
-        self.snippets = os.listdir(os.path.expanduser(self.minisnip_dir))
+        self.minisnip_dir = None
+        self.snippets = []
 
     def gather_candidates(self, context):
-        """Returns all snippets in the users
-        vim minisnip directory"""
-        filetype = '_' + context['filetype'] + '_'
+        if not self.minisnip_dir:
+            self.minisnip_dir = self.vim.eval('g:minisnip_dir')
+            self.snippets = os.listdir(os.path.expanduser(self.minisnip_dir))
+
+        ft = context['filetype']
+        ft_reg = '^_(' + format(ft.replace('.', '|')) + ')_'
```

Comment:
Can you add a comment to this line explaining what and why you're doing this?

Reply:
Done.